src/main/java/com/example/book/service/BookServiceImpl.java

```diff
@@ -50,6 +54,18 @@ public BookDto updateBook(Long id, BookDto updateRequest) {
         return bookMapper.toDto(bookRepository.save(existingBook));
     }
 
+    @Override
+    public List<BookDtoWithoutCategoryIds> findByCategoryId(Long id, Pageable pageable) {
+        if (!categoryRepository.existsById(id)) {
+            throw new EntityNotFoundException("Category with id " + id + " not found");
+        }
+        return bookRepository
+                .findByCategoryId(id, pageable)
+                .stream()
+                .map(bookMapper::toDtoWithoutCategories)
+                .toList();
```

Comment:
```suggestion
        return bookRepository
                .findByCategoryId(id, pageable)
                .map(bookMapper::toDtoWithoutCategories);
```
Return Page when you use pagination.